File: OOPS_LAB/Program_4/myString.cpp
#include<iostream>
using namespace std;

class MyString{
    private:
        string str;
    public:
        MyString()
        {
            str="";
        }

        MyString(string ipt)
        {
            str=ipt;
        }

        int StringLength()
        {
            int count=0;
            while (str[count] != '\0')
            {
                count++;
            }

            return count;
        }

        void CompareString(MyString other)
        {
            if (this->StringLength()==other.StringLength())
            {
                cout<<"String has equal length"<<endl;
            }
            else{
                cout<<"String does not have equal length"<<endl;
            }
        }
};

int main(int argc, char const *argv[])
{
    MyString *str = new MyString("Ashf");
    MyString *str2 = new MyString("Alam");
    cout<<str->StringLength();
    cout<<endl;
    str->CompareString(*str2);
    return 0;
}

File: OOPS_LAB/Program_4/myString.java
class StringO{
    private String str;

    public StringO(String ipt)
    {
        str=ipt;
    }

    public int findLength()
    {
        return str.length();
    }

    public void compare(StringO other)
    {
        if (this.str==other.str) {
            System.out.println("Both String are equals");
        }
        else{
            System.out.println("Both String are not equals");
        }
    }
}

class myString{
    public static void main(String[] args) {
        StringO str1 = new StringO("Ashfak");
        StringO str2 = new StringO("Alam");
        System.out.println(str1.findLength());
        str1.compare(str2);
    }
}